contactform.php

```diff
@@ -41,29 +53,187 @@ public function __construct()
+                        'type' => 'switch',
+                        'label' => $this->trans('Send confirmation email to your customers', array(), 'Modules.Contactform.Admin'),
+                        'desc' => $this->trans('Click Yes and your customers will receive a confirmation email without the message content', array(), 'Modules.Contactform.Admin'),
+                        'name' => self::SEND_CONFIRMATION_EMAIL,
+                        'is_bool' => true,
+                        'required' => true,
+                        'values' => array(
+                            array(
+                                'id' => self::SEND_CONFIRMATION_EMAIL . '_on',
+                                'value' => 1,
+                                'label' => $this->trans('Enabled', array(), 'Admin.Global')
+                            ),
+                            array(
+                                'id' => self::SEND_CONFIRMATION_EMAIL . '_off',
+                                'value' => 0,
+                                'label' => $this->trans('Disabled', array(), 'Admin.Global')
+                            )
+                        )
+                    ),
+                    array(
+                        'type' => 'switch',
+                        'label' => $this->trans("Receive customers' messages by email", array(), 'Modules.Contactform.Admin'),
+                        'desc' => $this->trans('By default,  you will only receive contact messages through your Customer service tab.', array(), 'Modules.Contactform.Admin'),
+                        'name' => self::SEND_NOTIFICATION_EMAIL,
+                        'is_bool' => true,
+                        'required' => true,
+                        'values' => array(
+                            array(
+                                'id' => self::SEND_NOTIFICATION_EMAIL . '_on',
+                                'value' => 1,
+                                'label' => $this->trans('Enabled', array(), 'Admin.Global')
+                            ),
+                            array(
+                                'id' => self::SEND_NOTIFICATION_EMAIL . '_off',
+                                'value' => 0,
+                                'label' => $this->trans('Disabled', array(), 'Admin.Global')
+                            )
+                        )
+                    )
+                ),
+                'submit' => array(
+                    'name' => self::SUBMIT_NAME,
+                    'title' => $this->trans('Save', array(), 'Admin.Actions'),
+                )
+            ),
+        );
+        $helper = new HelperForm();
+        $helper->table = $this->table;
+        $lang = new Language((int) Configuration::get('PS_LANG_DEFAULT'));
+        $helper->default_form_language = $lang->id;
+        $helper->submit_action = 'update-configuration';
+        $helper->currentIndex = $this->context->link->getAdminLink('AdminModules', false).'&configure='
```

Comment:
You should use `parse_url()` to make sure that there's a query string in the URL returned by `$this->context->link->getAdminLink('AdminModules', false)`, then `http_build_query()` to rebuild the query string with your arguments.

Reply:
not my code...

Reply:
done